dotnet/samples/HomeAutomation/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿/*
+ Copyright (c) Microsoft. All rights reserved.
+
+ Example application that uses Semantic Kernel.
+
+ Loads app configuration from:
+ - appsettings.json.
+ - appsettings.{Environment}.json.
+ - Secret Manager when the app runs in the "Development" environment (set through the DOTNET_ENVIRONMENT variable).
+ - Environment variables.
+ - Command-line arguments.
+*/
+
+using HomeAutomation.Plugins;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace HomeAutomation;
+
+internal static class Program
+{
+    internal static async Task Main(string[] args)
+    {
+        HostApplicationBuilder builder = Host.CreateApplicationBuilder(args);
+        builder.Services.AddHostedService<Worker>();
+        builder.Services.AddSingleton<AppConfiguration>();
+        builder.Services.AddSingleton<Kernel>(sp =>
+        {
+            AppConfiguration ac = sp.GetRequiredService<AppConfiguration>();
+            var builder = Kernel.CreateBuilder()
+                                // Add AI services here
+                                // Note that you can provide your custom http client when adding specific AI services
+                                .AddAzureOpenAIChatCompletion(ac.AzureOpenAiDeployment, ac.AzureOpenAiEndpoint, ac.AzureOpenAiApiKey);
+            // Add plugins to include in kernel here
+            builder.Plugins.AddFromType<MyTimePlugin>();
+            Kernel kernel = builder.Build();
+
+            return kernel;
+        });
+
+        using IHost host = builder.Build();
+
+        await host.RunAsync();
+    }
+}
+
+/// <summary>
+/// Actual code to run.
+/// </summary>
+internal sealed class Worker : BackgroundService
+{
+    private readonly IHostApplicationLifetime _hostApplicationLifetime;
+    private readonly Kernel _kernel;
+
+    public Worker(IHostApplicationLifetime hostApplicationLifetime, Kernel kernel)
+    {
+        _hostApplicationLifetime = hostApplicationLifetime;
+        _kernel = kernel;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
```

Comment:
Could we make this a more interactive experience (e.g., Console.Read) so that the developer can test out different prompts without needing to rewrite the code?

Reply:
If this is done, we should also add some test cases to make sure this doesn't break in the future. For example,

| Question                  | Required Plugins    |
| ------------------- | ------------------- |
| "What time is it?"    | Time |
| "Can you turn off the ceiling light in the kitchen?" | Light |
| "Can you turn off all the lights in the kitchen?" | Lights |
| "When does the sunset?" | Weather |
| "When does the sunset tomorrow? | Weather + Time |
| "Will you let me know when its 5:00?" | Schedular + Time |
| "Will you turn on the lights at 7:00?" | Schedular + Time + Light |
| "Will you turn on the lights when the sunsets?" | Schedular + Time + Light + Weather |

Reply:
Absolutely.  I'm just keeping it with static strings for now because I don't want to re-type stuff all the time until the app is finalized.  I'll switch it to read input when we're set.

Reply:
Questions have been suggested in app